### PR TITLE
87 write more end to end tests edge cases negative tests

### PR DESCRIPTION
**Pull request by nponsard**

### CODE REVIEW

File: .github/workflows/rust-tests.yml [hunk 1]
```diff
@@ -56,18 +50,38 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: install opensc and dependencies
-        run: apt-get update && apt-get install -y opensc curl libssl-dev build-essential pkg-config xxd jq
+        run: apt-get update && apt-get install -y opensc curl libssl-dev build-essential pkg-config xxd jq gnutls-bin
```
Comment (jans23) [inline .github/workflows/rust-tests.yml:53]: "apt" instead of "apt-get" is the recommended tool. :-)

Comment (nponsard) [reply .github/workflows/rust-tests.yml:53]: When running `apt` with no terminal/in scripts you get this warning : 

```
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
```

I prefer staying on the "stable" apt-get interface
